src/main/java/net/sf/jabref/logic/importer/fetcher/IsbnFetcher.java

```diff
@@ -66,28 +66,31 @@ public HelpFile getHelpPage() {
                 URL url = uriBuilder.build().toURL();
 
                 //Downloads the source code of the site and then creates a .bib file out of the String
-                URLDownload urlDownload = new URLDownload(url);
-                String bibtexString = urlDownload.downloadToString(StandardCharsets.UTF_8);
+                String bibtexString = Unirest.get(url.toString()).asString().getBody();
                 Optional<BibEntry> entry = BibtexParser.singleFromStringOptional(bibtexString, ImportFormatPreferences.fromPreferences(Globals.prefs));
 
                 if (entry.isPresent()) {
-                    BibEntry bibEntry = entry.get();
-                    if (bibEntry.hasField(FieldName.URL)) {
-                        bibEntry.clearField(FieldName.URL);
-                    }
-
-                    //Removes every non-digit character in the PAGETOTAL field.
-                    Optional<String> pagetotal = bibEntry.getFieldOptional(FieldName.PAGETOTAL);
-                    if (pagetotal.isPresent()) {
-                        bibEntry.setField(FieldName.PAGETOTAL, pagetotal.get().replaceAll("[\\D]", ""));
-                    }
-
-                    result = Optional.of(bibEntry);
+                    result = postProcessEntry(entry.get());
                 }
-            } catch (IOException | URISyntaxException e) {
+
+            } catch (UnirestException | IOException | URISyntaxException e) {
                 throw new FetcherException("Bad URL when fetching ISBN info", e);
             }
         }
         return result;
     }
+
+    private Optional<BibEntry> postProcessEntry(BibEntry entry) {
```

Comment:
I would return a BibEntry here as a BibEntry comes in and this is post processed. Always. The Optional is always present. Thus, the Optional conversion should be done by the caller.